internal/controllers/import_controller.go

```diff
@@ -83,10 +83,12 @@ func (r *CAPIImportReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		r.remoteClientGetter = remote.NewClusterClient
 	}
 
-	// TODO: we want the control plane initialized but removing for the time being
-	// capiPredicates := predicates.All(log, predicates.ClusterControlPlaneInitialized(log), predicates.ResourceHasFilterLabel(log, r.WatchFilterValue))
```

Comment:
This comment had a better handling of predicates. Specifically it used ResourceHasFilterLabel, which used label selector. This functionality is imported from controller-runtime and does not require to be tested, so this decreases the amount of code to review. Additionally this gives confidence in what is being added with the change. Why it is not being used here?

Reply:
This is the predicate covering 50% of what is being done within the custom logic we have - https://github.com/kubernetes-sigs/controller-runtime/blob/1449c365465511e047333f9bc504e47cb0fae1ae/pkg/predicate/predicate.go#L352

Reply:
That's not going to work in this case. We have to check the cluster label anyway in the predicate because we use the namespace as a fallback option: if the cluster doesn't have a label, look at the namespace. If we remove the label check for the cluster in the predicate, then when the namespace is not labeled, it's always going to return false.

Reply:
Yeah, maybe not in the current form. This requires a separate set of predicates on each resource to function properly, and ideally a mechanism to pass these predicates to event mapper.